src/code/sys_matrix.c

```diff
@@ -1,89 +1,1939 @@
+    temp2 = cmf->xy;
+    cmf->xw += temp1 * translation->x + temp2 * translation->y + cmf->xz * translation->z;
+    cmf->xx = temp1 * cos + temp2 * sin;
+    cmf->xy = temp2 * cos - temp1 * sin;
+
+    temp1 = cmf->yx;
+    temp2 = cmf->yy;
+    cmf->yw += temp1 * translation->x + temp2 * translation->y + cmf->yz * translation->z;
+    cmf->yx = temp1 * cos + temp2 * sin;
+    cmf->yy = temp2 * cos - temp1 * sin;
+
+    temp1 = cmf->zx;
+    temp2 = cmf->zy;
+    cmf->zw += temp1 * translation->x + temp2 * translation->y + cmf->zz * translation->z;
+    cmf->zx = temp1 * cos + temp2 * sin;
+    cmf->zy = temp2 * cos - temp1 * sin;
+
+    temp1 = cmf->wx;
+    temp2 = cmf->wy;
+    cmf->ww += temp1 * translation->x + temp2 * translation->y + cmf->wz * translation->z;
+    cmf->wx = temp1 * cos + temp2 * sin;
+    cmf->wy = temp2 * cos - temp1 * sin;
+
+    if (rot->y != 0) {
+        sin = Math_SinS(rot->y);
+        cos = Math_CosS(rot->y);
+
+        temp1 = cmf->xx;
+        temp2 = cmf->xz;
+        cmf->xx = temp1 * cos - temp2 * sin;
+        cmf->xz = temp1 * sin + temp2 * cos;
+
+        temp1 = cmf->yx;
+        temp2 = cmf->yz;
+        cmf->yx = temp1 * cos - temp2 * sin;
+        cmf->yz = temp1 * sin + temp2 * cos;
+
+        temp1 = cmf->zx;
+        temp2 = cmf->zz;
+        cmf->zx = temp1 * cos - temp2 * sin;
+        cmf->zz = temp1 * sin + temp2 * cos;
+
+        temp1 = cmf->wx;
+        temp2 = cmf->wz;
+        cmf->wx = temp1 * cos - temp2 * sin;
+        cmf->wz = temp1 * sin + temp2 * cos;
+    }
+
+    if (rot->x != 0) {
+        sin = Math_SinS(rot->x);
+        cos = Math_CosS(rot->x);
+
+        temp1 = cmf->xy;
+        temp2 = cmf->xz;
+        cmf->xy = temp1 * cos + temp2 * sin;
+        cmf->xz = temp2 * cos - temp1 * sin;
+
+        temp1 = cmf->yy;
+        temp2 = cmf->yz;
+        cmf->yy = temp1 * cos + temp2 * sin;
+        cmf->yz = temp2 * cos - temp1 * sin;
+
+        temp1 = cmf->zy;
+        temp2 = cmf->zz;
+        cmf->zy = temp1 * cos + temp2 * sin;
+        cmf->zz = temp2 * cos - temp1 * sin;
+
+        temp1 = cmf->wy;
+        temp2 = cmf->wz;
+        cmf->wy = temp1 * cos + temp2 * sin;
+        cmf->wz = temp2 * cos - temp1 * sin;
+    }
+}
+
+/**
+ * @brief Set current to a general translation and rotation using YXZ Tait-Bryan angles: T Ry Rx Rz -> current
+ *
+ * This means a (column) vector is first rotated around Y, then around X, then around Z, then translated, then gets
+ * transformed by whatever the matrix was previously.
+ *
+ * @param x amount to translate in X direction.
+ * @param y amount to translate in Y direction.
+ * @param z amount to translate in Z direction.
+ * @param rot vector of rotation angles.
+ *
+ * @remark original name appears to be "Matrix_softcv3_load"
+ */
+void Matrix_SetTranslateRotateYXZ(f32 x, f32 y, f32 z, Vec3s* rot) {
+    MtxF* cmf = sCurrentMatrix;
+    f32 sp30 = Math_SinS(rot->y);
+    f32 sp2C = Math_CosS(rot->y);
+    f32 sp28;
+    f32 sp24;
```

Comment:
```suggestion
    f32 sinY = Math_SinS(rot->y);
    f32 cosY = Math_CosS(rot->y);
    f32 cosX;
    f32 sinX;
```

Reply:
sp28 and sp24 are also used for Z, which is one reason I didn't name them.

Reply:
Ah didn't see that. In this case I would suggest something like `cosXZ` or `cosTemp`, but up to you.